projects/xrsh/programs/xrsh/tests/basic.nix

```diff
@@ -0,0 +1,32 @@
+{
+  sources,
+  ...
+}:
+
+{
+  name = "xrsh";
+
+  nodes = {
+    machine =
+      { ... }:
+      {
+        imports = [
+          sources.modules.ngipkgs
+          sources.modules.programs.xrsh
+          sources.examples.xrsh.basic
+        ];
+      };
+  };
+
+  testScript =
+    { nodes, ... }:
+    ''
+      start_all()
+
+      machine.succeed("xrsh >&2 &")
+
+      # xrsh serves defaultly on :8080
+      machine.wait_for_open_port(8090)
```

Comment:
@eljamm could we also `machine.wait_for_open_port(config.nodes.machine.config.programs.xrsh.port)`?